src/core/thread/mesh_forwarder.hpp

```diff
@@ -248,6 +248,28 @@ class MeshForwarder
         kDataRequestRetryDelay = 200,   ///< Retry delay in milliseconds (for sending data request if no buffer).
     };
 
+    enum
+    {
+        /** Maximum number of MAC layer tx attempts for an outbound direct frame.
+         *
+         */
+        kDirectFrameMaxMacTxAttempts   = OPENTHREAD_CONFIG_MAC_MAX_TX_ATTEMPTS_DIRECT_FRAME,
+
+        /** Maximum number of MAC layer tx attempts for an outbound indirect frame (for a sleepy child) after receiving
+         *  a data request command (data poll) from the child.
+         *
+         */
+        kIndirectFrameMaxMacTxAttempts = OPENTHREAD_CONFIG_MAC_MAX_TX_ATTEMPTS_INDIRECT_FRAME,
+
+        /** Maximum number of tx attempts by `MeshForwarder` for an outbound indirect frame (for a sleepy child). The
+         * `MeshForwader` attempts occur following the reception of a new data request command (a new data poll) from
+         * the sleepy child.
+         *
+         */
+        kIndirectFrameMaxDataPollTriggeredTxAttempts =
+            OPENTHREAD_CONFIG_MAX_DATA_POLL_TRIGGERED_TX_ATTEMPTS_INDIRECT_FRAME,
```

Comment:
nit: this is getting pretty long, but is inconsistent with the naming above.  The names above include `MAC` in the name.  How about shortening to something like:

- `OPENTHREAD_CONFIG_MAX_TX_ATTEMPTS_DIRECT`
- `OPENTHREAD_CONFIG_MAX_TX_ATTEMPTS_INDIRECT_PER_POLL`
- `OPENTHREAD_CONFIG_MAX_TX_ATTEMPTS_INDIRECT_POLLS`